nix/default.nix

```diff
@@ -84,6 +87,7 @@ let
         shellHook = ''
           export NODE_PATH="${nodeModules}/lib/node_modules"
           export PATH="${nodeModules}/bin:$PATH"
+          export NPM_CONFIG_PACKAGE_LOCK_ONLY="${if packageLockOnly then "true" else "false"}"
```

Comment:
Since this only takes effect when set to `true`, why not just leave it unset if `packageLockOnly` is `false`? You could do

```
${pkgs.lib.optionalString packageLockOnly "export NPM_CONFIG_PACKAGE_LOCK_ONLY=true"}
```

If you want to set it either way, you can also do

```
export NPM_CONFIG_PACKAGE_LOCK_ONLY="${pkgs.lib.boolToString packageLockOnly}"
```

Reply:
Thanks, completely forgot about `optionalString`